docs/articles/retrieval_augmented_generation_eval_qdrant_arize.md

```diff
@@ -0,0 +1,519 @@
+
+**relevance:** Are the retrieved documents grounded in the response? Q&A correctness: Are your application's responses grounded in the retrieved context?
+
+**hallucinations:** Is your application making up false information?
+
+We can do this as :
+
+```jsx
+queries_df = get_qa_with_reference(px.Client())
+retrieved_documents_df = get_retrieved_documents(px.Client())
+```
+
+Next, we define our evaluation model and evaluators.
+
+Evaluators are built on top of language models and prompt the LLM to assess the quality of responses, the relevance of retrieved documents, etc., and provide a quality signal even in the absence of human-labeled data.
+
+Pick an evaluator type and instantiate it with the language model you want to use to perform evaluations using our battle-tested evaluation templates. We can do this as :
+
+```python
+eval_model = OpenAIModel(
+    model="gpt-4-turbo-preview",
+)
+hallucination_evaluator = HallucinationEvaluator(eval_model)
+qa_correctness_evaluator = QAEvaluator(eval_model)
+relevance_evaluator = RelevanceEvaluator(eval_model)
+
+hallucination_eval_df, qa_correctness_eval_df = run_evals(
+    dataframe=queries_df,
+    evaluators=[hallucination_evaluator, qa_correctness_evaluator],
+    provide_explanation=True,
+)
+relevance_eval_df = run_evals(
+    dataframe=retrieved_documents_df,
+    evaluators=[relevance_evaluator],
+    provide_explanation=True,
+)[0]
+
+px.Client().log_evaluations(
+    SpanEvaluations(eval_name="Hallucination", dataframe=hallucination_eval_df),
+    SpanEvaluations(eval_name="QA Correctness", dataframe=qa_correctness_eval_df),
+)
+px.Client().log_evaluations(DocumentEvaluations(eval_name="Relevance", dataframe=relevance_eval_df))
+```
+
+The evaluations should now appear as annotations on the appropriate spans in Phoenix.
+
+**Key Observations**
+
+![../assets/use_cases/retrieval_augmented_generation_eval_qdrant_arize/naive_rag_run_evals.png](../assets/use_cases/retrieval_augmented_generation_eval_qdrant_arize/naive_rag_run_evals.png)
+
+This is the aggregated view where we can see `Total Traces` (which is essentially the total number of eval queries run) , `Total Tokens` processed , `Latency P50` , `Latency P99` , `Hallucination`  and `QA Correctness`. Along with Overall `Relevance` metrics like *nDCG , Precision and Recall.*
+
+For the given evaluations for Naive RAG we have : `Hallucination` = 18% and `QA Correctness` = 91%. Lets find out the problematic query/ies and address the issues to achieve our aim - **0% Hallucination and 100% QA Correctness**.
+
+Notice the `evaluations` for each eval questions which captures these for each query too which makes spotting the problematic queries instantly.
```

Comment:
`evaluations` for each eval question, capture these [these = what?] for each query [so for each eval question AND for each query - i.e., "query" is different than "question"?]

Reply:
Fixed

Reply:
@atarora
so "for each eval question" AND "for each query"?
 - i.e., "query" is different than "question"?